### PR TITLE
Pro 6290 locale switcher UI

### PR DESCRIPTION
**Pull request by ValJed**
[PRO-6290](https://linear.app/apostrophecms/issue/PRO-6290/[locale-switcher-epic]-as-a-user-i-want-to-switch-locales-from-within)

## Summary

Locale switcher UI:
* Extract agnostic locale switcher UI inside a pure UI vue component: `AposLocalePicker.vue`.
* Use this new component in `TheAposAdminBarLocale.vue`.
* Creates a vue component for the doc editor locale switcher `AposDocLocalePicker.vue`.
* Check for each locale if the current doc has been localized already (to show in the list the green or empty dots)
* Disables the locale switcher if current doc has an error
* If current doc has an error show confirmation modal asking to save the doc before switching or to discard
* Updates the confirmation modal by adding two options:
- One to show the close modal button that didn't exist until now
- One for the tiny mode to make padding match the designs

## What are the specific steps to test this change?

Editor modal should have the new locale switcher UI, with the confirmation modal working in changes have been made as well as a disable state if current doc has error.

## What kind of change does this PR introduce?

- [ ] Bug fix
- [X] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been update

### CODE REVIEW

File: modules/@apostrophecms/ui/ui/apos/components/AposLocalePicker.vue [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+const filteredLocales = computed(() => {
+  return locales.value.filter(({ name, label }) => {
+    const matches = term =>
+      term.toLowerCase().includes(search.value.toLowerCase());
+    return matches(name) || matches(label);
+  });
+});
+
+const availableLocales = computed(() => {
+  return locales.value.filter(locale => !!props.localized[locale.name]);
```
Comment (haroun) [inline modules/@apostrophecms/ui/ui/apos/components/AposLocalePicker.vue:89]: Why not use `isLocalized` from below?

File: modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue [hunk 2]
```diff
@@ -14,6 +14,17 @@
         @click="confirmAndCancel"
       />
     </template>
+    <template #localeDisplay>
```
Comment (haroun) [inline modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue:17]: Must be added to https://github.com/apostrophecms/doc-template-library/blob/fd44c2093c9277bbfc46e7ec945db4e142ab90a8/ui/apos/components/AposTemplateLibraryEditor.vue#L127

File: modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue [hunk 3]
```diff
@@ -168,7 +179,8 @@ export default {
         active: false,
         triggerFocusRefresh: 0,
         type: 'overlay',
-        showModal: false
+        showModal: false,
+        componentType: 'editorModal'
```
Comment (haroun) [inline modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue:183]: What is this?

**Review by haroun — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/ui/ui/apos/components/AposLocalePicker.vue (hunk 1 above)
Comment (ValJed) [reply modules/@apostrophecms/ui/ui/apos/components/AposLocalePicker.vue:89]: Changed in the next PR.

File: modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue (hunk 2 above)
Comment (ValJed) [reply modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue:17]: Adding a note to myself, but not sure we need a locale switcher for doc template libraries.

File: modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue (hunk 3 above)
Comment (ValJed) [reply modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue:183]: I have no idea lol, removed in the next PR.

**Review by haroun — APPROVED** [review]

**Pull request merged**